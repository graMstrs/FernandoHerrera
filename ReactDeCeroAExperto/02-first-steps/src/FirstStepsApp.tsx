import { ItemCounter } from "./shopping-cart/ItemCounter";


interface ItemInCart {
    productName: string;
    quantity: number;
}

const itemsInCart: ItemInCart[] =
    [
        { 
            productName: 'Nintendo switch 2', 
            quantity: 1
        },
        { 
            productName: 'Pro controller', 
            quantity: 4 
        },
        { 
            productName: 'Smash Bros', 
            quantity: 3 
        },
    ];

export function FirstStepsApp() {
    return (
        <>  
            <h1>Carrito de compras</h1>

            {
                itemsInCart.map( ({productName, quantity }) => (
                    <ItemCounter key={ productName } item={ productName } quantity={ quantity } />
                ))
            }

            
            { /*<ItemCounter 
                item="Nintendo switch 2" 
                quantity={1}
            />
            <ItemCounter 
                item="Pro controller" 
                quantity={4}
            />
            <ItemCounter 
                item="Smash Bros" 
                quantity={1}
            />*/ }
            
        </>
    )
}