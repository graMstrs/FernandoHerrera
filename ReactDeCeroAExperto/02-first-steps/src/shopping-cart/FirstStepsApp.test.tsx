import { describe, expect, test, vi } from "vitest";
import { render, screen } from "@testing-library/react";
import { FirstStepsApp } from "../FirstStepsApp";


vi.mock('./shopping-cart/ItemCounter', () => ({
    ItemCounter: () => <div data-testid="ItemCounter" />,
}));


describe('FirstStepssApp',() => {
    test('Should match snapshot', () => {
        const { container } = render(<FirstStepsApp />);
        
        expect(container).toMatchSnapshot();
    });

     test('Should render the correct number of ItemCounter components', () => {
        render (<FirstStepsApp />)

        const itemCounters = screen.getAllByTestId('ItemCounter');

        expect(itemCounters.length).toBe(3);
    });


});