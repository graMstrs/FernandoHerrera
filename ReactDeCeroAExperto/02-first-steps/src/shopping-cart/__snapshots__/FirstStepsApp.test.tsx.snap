// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`FirstStepssApp > Should match snapshot 1`] = `
<div>
  <h1>
    Carrito de compras
  </h1>
  <section
    class="item-row"
  >
    <span
      class="item-text"
      style="color: black;"
    >
      Nintendo switch 2
    </span>
    <button>
      +1
    </button>
    <span>
      1
    </span>
    <button>
      -1
    </button>
  </section>
  <section
    class="item-row"
  >
    <span
      class="item-text"
      style="color: black;"
    >
      Pro controller
    </span>
    <button>
      +1
    </button>
    <span>
      4
    </span>
    <button>
      -1
    </button>
  </section>
  <section
    class="item-row"
  >
    <span
      class="item-text"
      style="color: black;"
    >
      Smash Bros
    </span>
    <button>
      +1
    </button>
    <span>
      3
    </span>
    <button>
      -1
    </button>
  </section>
</div>
`;
